### PR TITLE
chore(DTFS2-6994): remove moment from portal-api tests

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
Moment.js is deprecated & should be replaced. This PR covers separate changes to tests, without changing behaviour - see #2691. The e2e test is covered by #2690 

## Resolution :heavy_check_mark:
Replace moment() with vanilla Dates where possible and date-fns where necessary

## Miscellaneous :heavy_plus_sign:
- rename and move unit tests labelled as api tests

## Note ⚠️ 
- need to remove package dependency once both PRs are merged

### CODE REVIEW

File: portal-api/api-tests/fixtures/deal-fully-completed-issued-and-unissued-facilities.js
Comment (AlexBramhill) [inline portal-api/api-tests/fixtures/deal-fully-completed-issued-and-unissued-facilities.js:5]:
```suggestion
const nowDate = Date.now()
```

File: portal-api/api-tests/v1/deals/deals-status-facilities.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/deals/deals-status-facilities.api-test.js:637]: reminder of this comment being here

File: portal-api/api-tests/v1/loan/loan-change-cover-start-date-validation.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/loan/loan-change-cover-start-date-validation.api-test.js:136]: You're calling `valueOf` twice

File: portal-api/src/v1/gef/controllers/validation/eligibilityCriteria.test.js
Comment (AlexBramhill) [inline portal-api/src/v1/gef/controllers/validation/eligibilityCriteria.test.js:5]: good spot

**Review by AlexBramhill — COMMENTED** [review]
can approve after the small comments -- Good job here :)

File: portal-api/api-tests/v1/bonds/bond-issue-facility-validation.api-test.js
Comment (MarRobSoftwire) [reply portal-api/api-tests/v1/bonds/bond-issue-facility-validation.api-test.js:13]: Doesnt work at 276 but have moved to 171

File: portal-api/src/v1/helpers/date.ts
Comment (MarRobSoftwire) [inline portal-api/src/v1/helpers/date.ts:1]: Taken from #2691 some functions aren't actually used in this PR

File: portal-api/api-tests/v1/deals/deals-status-submitted-updates-unsubmitted-issued-facilities.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/deals/deals-status-submitted-updates-unsubmitted-issued-facilities.api-test.js:12]: double line

```suggestion
```

File: portal-api/api-tests/v1/bonds/bonds-validation.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/bonds/bonds-validation.api-test.js:410]: Free new line


```suggestion
```

File: portal-api/api-tests/v1/bonds/bonds-validation.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/bonds/bonds-validation.api-test.js:505]: Free new line


```suggestion
```

File: portal-api/api-tests/v1/bonds/bonds-validation.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/bonds/bonds-validation.api-test.js:787]: Free new line

```suggestion
```

File: portal-api/api-tests/v1/loan/loan-change-cover-start-date-validation.api-test.js
Comment (AlexBramhill) [inline portal-api/api-tests/v1/loan/loan-change-cover-start-date-validation.api-test.js:397]:
```suggestion
```

File: portal-api/src/v1/helpers/date.ts
Comment (AlexBramhill) [inline portal-api/src/v1/helpers/date.ts:9]: To be the biggest pain in the world can we have tests for this please

**Review by AlexBramhill — COMMENTED** [review]
Really good - Few formatting things to address and to add some tests to date.ts, but otherwise I'm ready for this to go!

File: portal-api/src/v1/helpers/date.test.ts
Comment (AlexBramhill) [inline portal-api/src/v1/helpers/date.test.ts:1]: Could you look into the analysis errors here? I didn't notice in vscode!

File: portal-api/api-tests/fixtures/deal-fully-completed.js
Comment (MarRobSoftwire) [inline portal-api/api-tests/fixtures/deal-fully-completed.js:1]: This import isn't getting any codacy problems

File: portal-api/api-tests/v1/bonds/bonds.api-test.js
Comment (MarRobSoftwire) [inline portal-api/api-tests/v1/bonds/bonds.api-test.js:1]: No codacy problems

File: portal-api/unit.jest.config.js
Comment (AlexBramhill) [inline portal-api/unit.jest.config.js:2]: With the changes below -- can we remove this?

**Review by AlexBramhill — APPROVED** [review]
lgtm -- one comment that we can look at after this is merged

**Review by AlexBramhill — APPROVED** [review]

**Review by AlexBramhill — DISMISSED** [review]
lgtm

**Pull request merged**
